dist/src/main/java/io/camunda/tasklist/zeebe/ServiceTaskMutationAdapter.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.tasklist.zeebe;
+
+import io.camunda.authentication.tenant.TenantAttributeHolder;
+import io.camunda.security.auth.Authentication;
+import io.camunda.service.JobServices;
+import io.camunda.service.UserTaskServices;
+import io.camunda.tasklist.webapp.service.TaskMutationAdapter;
+import io.camunda.webapps.schema.entities.tasklist.TaskEntity;
+import io.camunda.zeebe.gateway.rest.RequestMapper;
+import java.util.List;
+import java.util.Map;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile("!standalone")
+public class ServiceTaskMutationAdapter implements TaskMutationAdapter {
+
+  private final UserTaskServices userTaskService;
+  private final JobServices jobService;
+
+  public ServiceTaskMutationAdapter(
+      final UserTaskServices userTaskService, final JobServices jobService) {
+    this.userTaskService = userTaskService;
+    this.jobService = jobService;
+  }
+
+  @Override
+  public void assignUserTask(final TaskEntity task, final String assignee) {
+    userTaskService
+        .withAuthentication(getAuthenticationWithAuthorizedTenant(task))
+        .assignUserTask(task.getKey(), assignee, "", true);
+  }
+
+  @Override
+  public void unassignUserTask(final TaskEntity task) {
+    userTaskService
+        .withAuthentication(getAuthenticationWithAuthorizedTenant(task))
+        .unassignUserTask(task.getKey(), "");
+  }
+
+  @Override
+  public void completeUserTask(final TaskEntity task, final Map<String, Object> variables) {
+    final var taskKey = task.getKey();
+    if (isJobBasedUserTask(task)) {
+      jobService
+          .withAuthentication(getAuthenticationWithAuthorizedTenant(task))
+          .completeJob(taskKey, variables, null);
+    } else {
+      userTaskService
+          .withAuthentication(getAuthenticationWithAuthorizedTenant(task))
+          .completeUserTask(taskKey, variables, "");
+    }
+  }
+
+  private Authentication getAuthenticationWithAuthorizedTenant(final TaskEntity task) {
+    // if multi-tenancy is enabled, then Tasklist already does the tenant-check,
+    // meaning we can assume that the user/client is at least authorized
+    // for tenant assigned to the given task.
+    TenantAttributeHolder.withTenantIds(List.of(task.getTenantId()));
+    return RequestMapper.getAuthentication();
```

Comment:
This will not work if we are outside a Request scope, for example when it is done in a job.
Today, we don't have this use-case though.

Reply:
As you said, the scope is tasklist or does tasklist async calls outside of a request?